pkg/controller/machine/controller.go

```diff
@@ -462,6 +470,25 @@ func (r *ReconcileMachine) patchFailedMachineInstanceAnnotation(machine *machine
 	return nil
 }
 
+func (r *ReconcileMachine) setClusterIDLabel(ctx context.Context, m *machinev1.Machine) error {
+	infra := &configv1.Infrastructure{}
+	infraName := client.ObjectKey{Name: globalInfrastuctureName}
+
+	if err := r.Client.Get(ctx, infraName, infra); err != nil {
+		return err
+	}
```

Comment:
This could be substituted with https://github.com/openshift/machine-api-operator/blob/60eb82286c4138f1810dfbc49a9dbe2f504d2a97/pkg/controller/vsphere/util.go#L51-L65

With added tests from `pkg/controller/machine/machine_controller_test.go` you'll essentially cover both.

Reply:
I don't want to import a function from vsphere utils, but making a shared utils package in future makes sense. Anyway, it's not the scope of this PR